### PR TITLE
[MM-60167] Migrate tooltips of "components/sidebar/sidebar_channel/sidebar_channel_link/sidebar_channel_link.tsx" to WithTooltip

### PR DESCRIPTION
**Pull request by Camillarhi**
#### Summary

- Migrate tooltips of `components/sidebar/sidebar_channel/sidebar_channel_link/sidebar_channel_link.tsx` to WithTooltip
- Add  `onEntering` as a prop to `Withtooltip` component
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/27929
Jira https://mattermost.atlassian.net/browse/MM-60167
#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/sidebar/sidebar_channel/sidebar_channel_link/sidebar_channel_link.tsx [hunk 1]
```diff
@@ -201,20 +200,15 @@ export default class SidebarChannelLink extends React.PureComponent<Props, State
...
             labelElement = (
-                <OverlayTrigger
-                    delayShow={Constants.OVERLAY_TIME_DELAY}
-                    placement='top'
-                    overlay={displayNameToolTip}
+                <WithTooltip
+                    id='channel-displayname__tooltip'
+                    title={label}
+                    placement={'top'}
                     onEntering={this.removeTooltipLink}
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/sidebar/sidebar_channel/sidebar_channel_link/sidebar_channel_link.tsx:208]: lets remove this onEntering prop

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/sidebar/sidebar_channel/sidebar_channel_link/sidebar_channel_link.tsx [hunk 2]
```diff
@@ -201,20 +200,14 @@ export default class SidebarChannelLink extends React.PureComponent<Props, State
...
-                <Tooltip id='channel-displayname__tooltip'>
-                    {label}
-                </Tooltip>
-            );
             labelElement = (
-                <OverlayTrigger
-                    delayShow={Constants.OVERLAY_TIME_DELAY}
-                    placement='top'
-                    overlay={displayNameToolTip}
-                    onEntering={this.removeTooltipLink}
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/sidebar/sidebar_channel/sidebar_channel_link/sidebar_channel_link.tsx:214]: lets remove `removeTooltipLink` unused function as well

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

**Comment by Camillarhi** [comment]
> This looks good :shipit: Thank you

Thanks

**Review by Rajat-Dabade — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**